third_party/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.19)
```

Comment:
We have CMake 3.12+ as min. requirements in our build instructions. We should decide if we really want to bump that version and if yes, we need to adjust our documentation accordingly.

Reply:
The top level project currently has `cmake_minimum_required(VERSION 3.16)`, and I was actually surprised that I found `cmake_minimum_required` declarations all over the project, since it's almost always set once, at the top level and that's it.  

Trying to revert to the `3.12` requirement as documented would probably be pretty tricky, although I can understand if you want avoid updating beyond 3.16, since out of the box Ubuntu 20.04 only supports up to 3.16 I think.  I do understand that as maintainers of an example repository it's important to strike a balance between wanting the features and improvements of more recent CMake versions and needing to support people who are pinned on older versions for whatever reason (usually the Android NDK CMake version in my experience).

Like I said, the only specific change that requires this version bump is CLI11, so if it's decided that it's undesirable to make this change I can remove that one.  For now I'll look try to bring the docs in alignment with this value.

Reply:
Yeah, it's kinda messy right now. I think we need to discuss this internally, as we have people compiling our samples on some very specific platforms. Not sure if those platforms do have access to recent CMake versions.